src/vs/workbench/browser/parts/editor/editorGroupsControl.ts

```diff
@@ -211,7 +211,8 @@ export class EditorGroupsControl implements IEditorGroupsControl, IVerticalSashL
 	}
 
 	private updateTabOptions(tabOptions: ITabOptions, refresh?: boolean): void {
-		const showTabCloseButton = this.tabOptions ? this.tabOptions.showTabCloseButton : false;
+		const tabCloseButton = this.tabOptions ? this.tabOptions.tabCloseButton : 'right';
+		const showTabCloseButton =  this.tabOptions ? this.tabOptions.showTabCloseButton : false;
```

Comment:
Just delete support for `showTabCloseButton` option since it is now supported via your `tabCloseButton` setting.